Code/Framework/AzCore/AzCore/Serialization/Json/JsonImporter.cpp

```diff
@@ -159,6 +160,311 @@ namespace AZ
         return ResultCode(Tasks::Import, Outcomes::Success);
     }
 
+    // Load the Imported Files and and Merge the JSON contents of the file
+    // into the target
+    auto JsonImportResolver::LoadAndMergeImportFile(rapidjson::Value& target, rapidjson::Document::AllocatorType& allocator,
+        const rapidjson::Value& importDirective, AZ::IO::PathView importAbsPath,
+        const AZ::StackedString& element, const JsonSerializationResult::JsonIssueCallback& issueReporter)
+        -> JsonSerializationResult::ResultCode
+    {
+        using ReporterString = AZStd::fixed_string<1024>;
+
+        if (auto importedObject = JsonSerializationUtils::ReadJsonFile(importAbsPath.Native());
+            importedObject.IsSuccess())
+        {
+            JsonApplyPatchSettings applyPatchSettings;
+            applyPatchSettings.m_reporting = issueReporter;
+
+            rapidjson::Value& importedDoc = importedObject.GetValue();
+            if (importDirective.IsObject())
+            {
+                auto patchField = importDirective.FindMember("patch");
+                if (patchField != importDirective.MemberEnd())
+                {
+                    JsonSerialization::ApplyPatch(importedDoc, allocator, patchField->value,
+                        AZ::JsonMergeApproach::JsonMergePatch, applyPatchSettings);
```

Comment:
this part is a little confusing to me, is the scenario where you use this when you have a file with an import like this?

 `test.setreg` 
```json
{
   "$import":{
       "patch":"somefile.setreg"
   }
   ...
}
```

So first it merges the patch from `somefile.setreg` into `test.setreg` and then merge the result into `target`?

Reply:
Well the "patch" isn't a file, but a JSON string.
So it merges the patch data into the current JSON file that is being imported and then merges imported JSON to the target.

Reply:
Cool thanks!